### PR TITLE
Storage management

### PR DESCRIPTION
**Pull request by FroVolod**
https://nomicon.io/Standards/StorageManagement

### CODE REVIEW

File: src/commands/account/add_key/autogenerate_new_keypair/save_keypair_to_keychain/mod.rs [hunk 1]
```diff
@@ -35,34 +35,36 @@ impl SaveKeypairToKeychainContext {
...
                 move |_signed_transaction, network_config, storage_message| {
                     let key_pair_properties_buf = serde_json::to_string(&item.key_pair_properties)?;
                     *storage_message = crate::common::save_access_key_to_keychain(
                         network_config.clone(),
-                        item.config.credentials_home_dir.clone(),
+                        credentials_home_dir.clone(),
                         &key_pair_properties_buf,
                         &item.key_pair_properties.public_key_str,
-                        &item.signer_account_id,
+                        &signer_account_id,
```
Comment (frol) [inline src/commands/account/add_key/autogenerate_new_keypair/save_keypair_to_keychain/mod.rs:67]: I would use `signed_transaction.signer_id` to make it more reliable and avoid unnecessary code (`let signer_account_Id` at the beginning of the function)

P.S. Apply the same reasoning to other places.

File: src/commands/account/delete_key/mod.rs [hunk 2]
```diff
@@ -33,21 +33,25 @@ impl DeleteKeyCommandContext {
 
 impl From<DeleteKeyCommandContext> for crate::commands::ActionContext {
     fn from(item: DeleteKeyCommandContext) -> Self {
+        let on_after_getting_network_callback: crate::commands::OnAfterGettingNetworkCallback =
+            std::sync::Arc::new(move |_network_config| {
+                Ok(crate::commands::PrepopulatedTransaction {
+                    signer_id: item.owner_account_id.clone().into(),
+                    receiver_id: item.owner_account_id.clone().into(),
```
Comment (frol) [inline src/commands/account/delete_key/mod.rs:40]: Do we need `.into()` here? In other places, you don't use it.

File: src/commands/account/storage_management/mod.rs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        let get_contract_account_id: GetContractAccountID =
+            std::sync::Arc::new(move |_network_config| Ok(contract_account_id.clone().into()));
+        Ok(Self {
+            config: previous_context.0,
+            get_contract_account_id,
+        })
+    }
+}
+
+pub type GetContractAccountID = std::sync::Arc<
```
Comment (frol) [inline src/commands/account/storage_management/mod.rs:38]: We use `Id` (with lower-case `d`) in other places, so let's rename it for consistency:

```suggestion
pub type GetContractAccountId = std::sync::Arc<
```

File: src/commands/account/storage_management/storage_withdraw.rs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                    receiver_id: get_contract_account_id(network_config)?,
+                    actions: vec![near_primitives::transaction::Action::FunctionCall(
+                        near_primitives::transaction::FunctionCallAction {
+                            method_name: "storage_withdraw".to_string(),
+                            args: serde_json::json!({
+                                "amount": previous_context.amount.clone().to_yoctonear().to_string()
+                            })
+                            .to_string()
+                            .into_bytes(),
+                            gas: crate::common::NearGas::from_str("300 TeraGas")
```
Comment (frol) [inline src/commands/account/storage_management/storage_withdraw.rs:71]: Let's try to use `50 TeraGas`

File: src/commands/account/storage_management/storage_withdraw.rs [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+                                .unwrap()
+                                .inner,
+                            deposit: crate::common::NearBalance::from_str("1 yoctoNear")
+                                .unwrap()
+                                .to_yoctonear(),
```
Comment (frol) [inline src/commands/account/storage_management/storage_withdraw.rs:76]:
```suggestion
                            deposit: crate::common::NearBalance::from_yoctonear(1).to_yoctonear(),
```

**Review by frol — APPROVED** [review]
Overall, looks good to me. Have you tested it with `near-social`?

File: src/commands/account/storage_management/storage_withdraw.rs (hunk 4 above)
Comment (FroVolod) [reply src/commands/account/storage_management/storage_withdraw.rs:71]: Done

File: src/commands/account/delete_key/mod.rs (hunk 2 above)
Comment (FroVolod) [reply src/commands/account/delete_key/mod.rs:40]: Done

File: src/commands/account/add_key/autogenerate_new_keypair/save_keypair_to_keychain/mod.rs (hunk 1 above)
Comment (FroVolod) [reply src/commands/account/add_key/autogenerate_new_keypair/save_keypair_to_keychain/mod.rs:68]: Done

**Pull request merged**